src/DynamoCore/Library/TypedParameter.cs

```diff
@@ -13,22 +26,17 @@ public class TypedParameter
     {
         private string summary;
 
-        public TypedParameter(string parameter, ProtoCore.Type type, object defaultValue = null)
-            : this(null, parameter, type, defaultValue) { }
-
-        public TypedParameter(
-            FunctionDescriptor function, string name, ProtoCore.Type type, object defaultValue = null)
+        public TypedParameter(TypedParameterParams parameter)
         {
-            if (name == null) 
-                throw new ArgumentNullException("name");
+            if (string.IsNullOrEmpty(parameter.ParameterName))
+                throw new ArgumentException("Invalid ParameterName");
 
-            Name = name;
-            Type = type;
-            DefaultValue = defaultValue;
-            Function = function;
+            Name = parameter.ParameterName;
+            Type = parameter.Type;
+            DefaultValue = parameter.DefaultValue;
         }
 
-        public FunctionDescriptor Function { get; set; }
+        public FunctionDescriptor Function { get; private set; }
```

Comment:
Yep which is why I highlighted why "this will be apparent when the new `PathManager` arrives (we need to pass in an optional `IPathManager` when `Function` is updated, so it needs to be a method instead of property setter).